flask_dance/contrib/slack.py

```diff
@@ -56,19 +57,27 @@ def make_slack_blueprint(
         storage: A token storage class, or an instance of a token storage
                 class, to use for this blueprint. Defaults to
                 :class:`~flask_dance.consumer.storage.session.SessionStorage`.
+        subdomain (str, optional): the name of the subdomain under which your
+            Slack space is accessed. Providing this may improve the login experience.
 
     :rtype: :class:`~flask_dance.consumer.OAuth2ConsumerBlueprint`
     :returns: A :ref:`blueprint <flask:blueprints>` to attach to your Flask app.
     """
     scope = scope or ["identify", "chat:write:bot"]
+    if subdomain:
+        hostname = "{subdomain}.slack.com".format(subdomain=subdomain)
+    else:
+        hostname = "slack.com"
     slack_bp = SlackBlueprint(
         "slack",
         __name__,
         client_id=client_id,
         client_secret=client_secret,
         scope=scope,
         base_url="https://slack.com/api/",
-        authorization_url="https://slack.com/oauth/authorize",
+        authorization_url="https://{hostname}/oauth/authorize".format(
```

Comment:
You can do this as `authorization_url = "https://slack.com/oauth/authorize" if subdomain is None else "https://{subdomain}.slack.com/oauth/authorize".format(subdomain=subdomain)`.

Reply:
Makes sense. Applied changes in latest commits.